src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1SpaceBookingService.kt

```diff
@@ -89,12 +89,16 @@ class Cas1SpaceBookingService(
     premises!!
     placementRequest!!
 
-    when (
-      val existingBooking =
-        cas1SpaceBookingRepository.findByPremisesIdAndPlacementRequestId(premisesId, placementRequestId)
-    ) {
-      null -> {}
-      else -> return existingBooking.id hasConflictError "A Space Booking already exists for this premises and placement request"
+    placementRequest.booking?.let {
+      if (it.isActive()) {
+        return it.id hasConflictError "A legacy Booking already exists for this premises and placement request"
+      }
+    }
+
+    cas1SpaceBookingRepository.findByPremisesIdAndPlacementRequestId(premisesId, placementRequestId)?.let {
```

Comment:
I think this will throw an exception if we have multiple cancelled space bookings linked to this placement reuqest. Probably best to a return a list and do something like

```
if(cas1SpaceBookingRepository.findByPremisesIdAndPlacementREquestId(premiseId, placementRequestId).any {  it.isActive }) {
   return error
}
```